libnetwork/iptables/firewalld.go

```diff
@@ -159,13 +200,27 @@ func OnReloaded(callback func()) {
 }
 
 // Call some remote method to see whether the service is actually running.
-func checkRunning() bool {
+func checkRunning() (bool, error) {
+
+	var err error
+
 	if connection == nil {
-		return false
+		return false, nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), dbusConnectionTimeout)
+	defer cancel()
+
+	var currentState string
+	err = connection.sysObj.CallWithContext(ctx, "org.freedesktop.DBus.Properties.Get", 0, dbusInterface, "state").Store(&currentState)
+
+	if err == nil {
+
+		// We are supposed to make sure firewalld is not in an INIT state
+		return (currentState == "RUNNING" || currentState == "FAILED"), nil
+	} else {
+		return false, fmt.Errorf("Failed to get current state of DBus connection: %v", err)
```

Comment:
1. [Error strings should not be capitalized.](https://go.dev/wiki/CodeReviewComments#error-strings)
2. I don't think that error message is accurate. The property being requested is the state _of FirewallD_. The usage of D-Bus is effectively an implementation detail.
3. [Wrap the error](https://go.dev/blog/go1.13-errors) so that the caller can introspect the cause of the failure if they cared about it.